node/src/accountant/scanners.rs

```diff
@@ -97,6 +100,56 @@ pub enum BeginScanError {
     CalledFromNullScanner, // Exclusive for tests
 }
 
+impl BeginScanError {
+    pub fn handle_error(
+        &self,
+        logger: &Logger,
+        scan_type: ScanType,
+        is_externally_triggered: bool,
+    ) {
+        match self {
+            BeginScanError::NothingToProcess => {
+                let log_message =
+                    format!("There was nothing to process during {:?} scan.", scan_type);
+
+                BeginScanError::log(logger, log_message, is_externally_triggered);
+            }
+            BeginScanError::ScanAlreadyRunning(timestamp) => {
+                let log_message = format!(
+                    "{:?} scan was already initiated at {}. \
+                     Hence, this scan request will be ignored.",
+                    scan_type,
+                    BeginScanError::timestamp_as_string(&timestamp)
+                );
+
+                BeginScanError::log(logger, log_message, is_externally_triggered);
+            }
+            BeginScanError::CalledFromNullScanner => {
+                if !cfg!(test) {
+                    panic!("Null Scanner shouldn't be running inside production code.")
+                }
+            }
+        }
+    }
+
+    fn timestamp_as_string(timestamp: &SystemTime) -> String {
+        const TIME_FORMATTING_STRING: &str =
+            "[year]-[month]-[day] [hour]:[minute]:[second].[subsecond digits:3]";
+
+        let offset_date_time = OffsetDateTime::from(*timestamp);
+        offset_date_time
+            .format(&parse(TIME_FORMATTING_STRING).unwrap())
+            .unwrap()
+    }
+
+    fn log(logger: &Logger, log_message: String, is_externally_triggered: bool) {
+        match is_externally_triggered {
+            true => info!(logger, "{}", log_message),
+            false => debug!(logger, "{}", log_message),
+        }
+    }
+}
+
 pub struct ScannerCommon {
     initiated_at_opt: Option<SystemTime>,
     pub payment_thresholds: Rc<PaymentThresholds>,
```

Comment:
Again, I cannot write this comment to the right place but please move the fn `update_timestamp_and_log` from `scanner_tools.rs` over here. 
Plus rename it, plus change the signature slightly. All is shown here:
```

impl ScannerCommon {
    fn new(payment_thresholds: Rc<PaymentThresholds>) -> Self {
        Self {
            initiated_at_opt: None,
            payment_thresholds,
        }
    }

    fn zero_timestamp_out_and_log(
        &mut self,
        scan_type: ScanType,
        logger: &Logger,
    ) {
        match self.initiated_at_opt.take() {
            Some(timestamp) => {
                let elapsed_time = SystemTime::now()
                    .duration_since(timestamp)
                    .expect("Unable to calculate elapsed time for the scan.")
                    .as_millis();
                info!(
                    logger,
                    "The {:?} scan ended in {}ms.", scan_type, elapsed_time
                );
            }
            None => {
                error!(
                    logger,
                    "Called scan_finished() for {:?} scanner but timestamp was not found",
                    scan_type
                );
            }
        };
    }
}
```